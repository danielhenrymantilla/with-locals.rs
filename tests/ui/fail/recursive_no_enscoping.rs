include!("../prelude.rs");

struct Implementor;

impl Implementor {
    #[with(recursive = true)]
    fn foo (&self)
      -> &'ref ()
    {
        &()
    }

    #[with(recursive = true)]
    fn bar (self: &'_ Self)
      -> &'ref ()
    {
        &()
    }
}

trait Trait {
    #[with(recursive = true)]
    fn foo (&self)
      -> &'ref ()
    {
        &()
    }

    #[with(recursive = true)]
    fn bar (self: &'_ Self)
      -> &'ref ()
    {
        &()
    }
}
