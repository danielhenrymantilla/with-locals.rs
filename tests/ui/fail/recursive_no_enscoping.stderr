error: Missing `#[with]` on the enscoping `impl` or `trait` block
  --> $DIR/recursive_no_enscoping.rs:29:5
   |
29 |     #[with(recursive = true)]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

error: Missing `#[with]` on the enscoping `impl` or `trait` block
  --> $DIR/recursive_no_enscoping.rs:22:5
   |
22 |     #[with(recursive = true)]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

error: Missing `#[with]` on the enscoping `impl` or `trait` block
  --> $DIR/recursive_no_enscoping.rs:13:5
   |
13 |     #[with(recursive = true)]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

error: Missing `#[with]` on the enscoping `impl` or `trait` block
 --> $DIR/recursive_no_enscoping.rs:6:5
  |
6 |     #[with(recursive = true)]
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^
